modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuildControllerTest.java

```diff
@@ -179,10 +181,43 @@ void testStopBuildIndexesOnChangePrimaryReplica() {
         verify(indexBuilder).stopBuildingIndexes(tableId(), PARTITION_ID);
     }
 
+    @Test
+    void testStartBuildIndexesOnPrimaryReplicaElectedOnlyForRegisteredIndexes() {
+        createIndex(INDEX_NAME);
+
+        int indexId = indexId(INDEX_NAME);
+
+        makeIndexAvailable(indexId);
+
+        setPrimaryReplicaWitchExpireInOneSecond(PARTITION_ID, NODE_NAME, clock.now());
+
+        verify(indexBuilder, never()).scheduleBuildIndex(
```

Comment:
When verifying that something does NOT happen, probably, it makes sense to use as wide matchers as possible (that is, `any()`/`anyInt()`/... instead of `eq(tableId())` and so on) to make sure we don't miss an invocation because we made a mistake in the matchers.

Reply:
I understand your proposal, but in my case a wide matchers check will not work.